transaction/src/manifest/lexer.rs

```diff
@@ -195,56 +172,69 @@ impl Lexer {
                 '1' => match self.advance()? {
                     '2' => match self.advance()? {
                         '8' => self.parse_int(&s, "i128", TokenKind::I128Literal),
-                        _ => Err(self.unexpected_char()),
+                        _ => Err(self.unexpected_char_previous()),
```

Comment:
Instead of all these `self.unexpected_char_previous`, I think it would be clearer if we:
* Captured a `let literal_start = self.current` at the top
* Captured a `let type_start = self.current` at `// type`
* In the `// integer` section, return a `LexerError::invalid_integer_literal(start, end)` with `LexerErrorKind::InvalidIntegerLiteral` and `start: literal_start` up to `end: self.current`.
* In the `// type` section, returne a `LexerError::invalid_integer_type(start, end)` with `LexerErrorKind::InvalidIntegerType` and `start: type_start` up to `end: self.current`.

Reply:
Ack

Reply:
Done